tools/component-extractor/src/extract.js

```diff
@@ -37,75 +57,94 @@ const outputDirectory = path.join(
-  try {
-    reactAPI = reactDocgen.parse(
-      Buffer.from(src, 'utf8'),
-      undefined,
-      undefined,
-      { filename: path.parse(componentPath).base },
+components
+  .filter((componentPath) => {
+    const shouldIgnore = IGNORED_COMPONENTS.every(
+      (ignored) => !componentPath.endsWith(ignored),
     );
-  } catch (err) {
-    console.log('Error parsing src for', componentPath);
-    throw err;
-  }
 
-  reactAPI.name = path.parse(componentPath).name;
-  reactAPI.importName = path.parse(componentPath).name;
-  reactAPI.styles = styles;
-  reactAPI.filename = componentPath.replace(rootDirectory, '');
-  reactAPI.importPath = `@material-ui/${muiSrc}`;
-
-  // Inheritance
-  const testInfo = await parseTest(reactAPI.filename, muiSrc);
-  const inheritance = getInheritance(testInfo, src);
-  reactAPI.inheritsFrom = inheritance ? inheritance.component : '';
-
-  if (typeof reactAPI.props.classes !== 'undefined') {
-    reactAPI.props.classes.flowType = {
-      name: 'classes',
-      elements: reactAPI.styles.classes,
+    if (!shouldIgnore) {
+      console.warn(`Ignoring ${componentPath}`);
+    }
+    return shouldIgnore;
+  })
+  .forEach(async (componentPath) => {
+    let component;
+    let src;
+    try {
+      src = readFileSync(componentPath, 'utf8');
+
+      if (
+        src.match(/@ignore - internal component\./) ||
+        src.match(/@ignore - do not document\./)
+      ) {
+        return;
+      }
+      component = await import(componentPath);
+    } catch (err) {
+      console.error(err.message);
+      return;
+    }
+    const styles = {
+      classes: [],
+      name: null,
     };
-  }
 
-  ensureExists(outputDirectory, 0o744, (err) => {
-    if (err) {
-      console.log('Error creating directory', outputDirectory);
-      console.log(err);
-      return;
+    if (component.styles && component.default.options) {
+      // Collect the customization points of the `classes` property.
+      styles.classes = Object.keys(
+        getStylesCreator(component.styles).create(theme),
+      ).filter((className) => !className.match(/^(@media|@keyframes)/));
+      styles.name = component.default.options.name;
     }
 
-    writeFileSync(
-      path.resolve(outputDirectory, `${kebabCase(reactAPI.name)}.json`),
-      JSON.stringify(reactAPI),
-    );
-    console.log('Extracted JSON for', componentPath);
+    let reactAPI;
+    const parsedPath = path.parse(componentPath);
+    try {
+      reactAPI = reactDocgen.parse(
+        Buffer.from(src, 'utf8'),
+        undefined,
+        undefined,
+        { filename: parsedPath.base },
+      );
+    } catch (err) {
+      console.error('Error parsing src for', componentPath);
+      console.error(err.message);
+      throw err;
+    }
+
+    reactAPI.name = parsedPath.name;
+    reactAPI.importName = parsedPath.name;
+    reactAPI.styles = styles;
+    reactAPI.filename = componentPath.replace(rootDirectory, '');
+    reactAPI.importPath = `@mui/${muiSrc}`;
```

Comment:
opportunity for future improvement: https://mui.com/material-ui/guides/minimizing-bundle-size/#option-one-use-path-imports